packages/teams-js/src/public/media.ts

```diff
@@ -350,12 +367,125 @@ export namespace media {
    */
   export interface AudioProps {
     /**
-     * Optional; the maximum duration in minutes after which the recording should terminate automatically.
+     * Optional; the maximum duration in minutes after which the recording should terminate automatically
      * Default value is defined by the platform serving the API.
      */
     maxDuration?: number;
   }
 
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Base class which holds the callback and notifies events to the host client
+   */
+  abstract class MediaController<T> {
+    protected controllerCallback: T;
+
+    public constructor(controllerCallback?: T) {
+      this.controllerCallback = controllerCallback;
+    }
+
+    protected abstract getMediaType(): MediaType;
+
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * This function will be implemented by the respective media class which holds the logic
+     * of specific events that needs to be notified to the app.
+     * @param mediaEvent indicates the event signed by the host client to the app
+     */
+    protected abstract notifyEventToApp(mediaEvent: MediaControllerEvent): void;
+
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * Function to notify the host client to programatically control the experience
+     * @param mediaEvent indicates what the event that needs to be signaled to the host client
+     */
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent): Promise<void> {
+      return new Promise<void>(resolve => {
+        ensureInitialized(FrameContexts.content, FrameContexts.task);
+        const err = isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
+        if (err) {
+          throw err;
+        }
+
+        const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
+        resolve(sendAndHandleSdkError('media.controller', [params]));
+      });
+    }
+
+    /**
+     * Function to programatically stop the ongoing media event
+     */
+    public stop(): Promise<void> {
+      return this.notifyEventToHost(MediaControllerEvent.StopRecording);
+    }
+  }
+
+  /**
+   * Callback which will register your app to listen to lifecycle events during the video capture flow
+   */
+  export interface VideoControllerCallback {
+    onRecordingStarted?(): void;
+  }
+
+  /**
+   * VideoController class is used to communicate between the app and the host client during the video capture flow
+   */
+  export class VideoController extends MediaController<VideoControllerCallback> {
+    protected getMediaType(): MediaType {
+      return MediaType.Video;
+    }
+
+    public notifyEventToApp(mediaEvent: MediaControllerEvent): void {
+      if (!this.controllerCallback) {
```

Comment:
Won't it be nice to have an overloaded/only constructor to assign the controllerCallback? (Follow up on this issue with v1 teams)

Reply:
To get unblocked, tabling this issue. Will create a work item.